app/src/main/java/com/github/openwebnet/repository/impl/EnvironmentRepositoryImpl.java

```diff
@@ -102,4 +106,19 @@ public EnvironmentRepositoryImpl() {
         });
     }
 
+    @Override
+    public Observable<Void> delete(Integer id) {
+        return Observable.create(subscriber -> {
+            try {
```

Comment:
I agree, this isn't ideal. Unfortunately it seems that Realm does **not** support "cascade delete" for now.

Anyway, even if it were supported I'm not sure it would work here because we don't link ```LightModel``` with ```EnvironmentModel``` but with ```environmentId``` (same for ```DeviceModel``` & ```AutomationModel```).

We could always use reflection on ```DomoticModel``` to find derived classes, but I don't think it would be a better implementation.
Another solution would be to use a compiler hook to generate a list of classes at compile-time but that seems like **a lot** of work for a *not-very-frequent* use case. Indeed, I know you would like to add few new device types (IP cam, temperature...) but once we have those there should'nt be any new device types.